lib/std/meta/trait.zig

```diff
@@ -490,6 +490,22 @@ pub fn hasUniqueRepresentation(comptime T: type) bool {
 
         .Pointer => |info| return info.size != .Slice,
 
+        .Union => |info| {
+            // Only packed and extern unions have guaranteed in-memory representation
+            switch (info.layout) {
+                else => return false,
+
+                .Extern, .Packed => {
+                    inline for (info.fields) |field| {
+                        const FieldType = field.field_type;
+                        if (comptime !hasUniqueRepresentation(FieldType)) return false;
+                    }
+                    return true;
```

Comment:
Unless all fields have the same bit size the union will contain padding bits for some fields, I think it is ok to just leave this as a false.

Reply:
Gotcha, I just did this.